python_modules/dagster/dagster/_core/host_representation/external_data.py

```diff
@@ -1604,8 +1604,18 @@ def external_resource_data_from_def(
     resource_type_def = resource_def
     if isinstance(resource_type_def, ResourceWithKeyMapping):
         resource_type_def = resource_type_def.wrapped_resource
+
     resource_type = str(type(resource_type_def))[8:-2]
 
+    if isinstance(
+        resource_type_def,
+        (
+            ConfigurableResourceFactoryResourceDefinition,
+            ConfigurableIOManagerFactoryResourceDefinition,
+        ),
+    ):
+        resource_type = str(resource_type_def.configurable_resource_cls)[8:-2]
```

Comment:
looks like code predates you, but what is this magic string mangling? Suggest extracting out to a function with a meaningful name for documentation purposes